pkg/apis/pipeline/v1beta1/task_validation_test.go

```diff
@@ -436,6 +436,48 @@ func TestTaskSpecValidate(t *testing.T) {
 	}
 }
 
+func TestTaskSpecValidateWithoutMutating(t *testing.T) {
+	type fields struct {
+		Params []v1beta1.ParamSpec
+		Steps  []v1beta1.Step
+	}
+	tests := []struct {
+		name          string
+		fields        fields
+		expectedError apis.FieldError
+	}{{
+		name: "empty param type",
+		fields: fields{
+			Params: []v1beta1.ParamSpec{{
+				Name:        "validparam",
+				Type:        v1beta1.ParamTypeString,
+				Description: "parameter",
+				Default:     v1beta1.NewArrayOrString("default"),
+			}, {
+				Name:        "param-with-invalid-type",
+				Description: "invalidtypedesc",
+				Default:     v1beta1.NewArrayOrString("default"),
+			}},
+			Steps: validSteps,
+		},
+		expectedError: apis.FieldError{
+			Message: `invalid value: `,
+			Paths:   []string{"params.param-with-invalid-type.type"},
+		},
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := &v1beta1.TaskSpec{
+				Params: tt.fields.Params,
+				Steps:  tt.fields.Steps,
+			}
+			if err := ts.Validate(context.Background()); err != nil {
+				t.Errorf("TaskSpec.Validate() = %v", err)
+			}
+		})
```

Comment:
In the test, can we compare the returned v.s. expected error like this [example](https://github.com/tektoncd/pipeline/blob/a4f9216e8957e52ce9bdf4af82cfd82148659539/pkg/apis/pipeline/v1beta1/task_validation_test.go#L1304-L1311) rather than just check if err is nil?